clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -185,9 +215,29 @@ public CompletableFuture start() throws Exception {
             return transport.send(handshake).thenRun(() -> {
                 hubConnectionStateLock.lock();
                 try {
+                    resetServerTimeout();
                     hubConnectionState = HubConnectionState.CONNECTED;
                     connectionState = new ConnectionState(this);
                     logger.log(LogLevel.Information, "HubConnected started.");
+
+                    this.pingTimer = new Timer();
+                    this.pingTimer.schedule(new TimerTask() {
+                        @Override
+                        public void run() {
+                            try {
+                                if (nextServerTimeout < System.currentTimeMillis()) {
```

Comment:
Super nit: I feel like this reads much better if you write it as  `System.currentTimeMillis() > nextServerTimeout`.
The check below as well.